grouper/entities/permission_grant.py

```diff
@@ -23,3 +23,15 @@
         ("grant_id", Optional[int]),
     ],
 )
+
+# Represents all grants of a specific permission, as a dictionary of users and service accounts who
+# have been granted that permission to lists of all the arguments to that permission that they have
+# been granted.
+AllGrantsOfPermission = NamedTuple(
```

Comment:
One thing that I think is a bit confusing -- but that I don't have any suggestion on how to improve -- is the use of "grant" to both mean the details of the grant itself, as well as to just represent that a user/service account/group has a permission. So under the former definition, a user with the same (permission, argument) through two different groups has two grants, whereas under the latter it only has one.

Reply:
This is an interesting point. I thought about it for a bit and then renamed this to `UniqueGrantsOfPermission`. Hopefully the "unique" part will hint at that difference: this isn't every grant, just the ones that provide unique permissions.